packages/vite/src/node/config.ts

```diff
@@ -1063,6 +1063,12 @@ export async function loadConfigFromFile(
   }
 }
 
+function getCurrentNodeTarget(): string {
+  // `process.versions.node` is usually in the form of `22.2.0`, but nightly builds are in the form of `23.0.0-nightly202405252079a7aec4`.
+  const version = process.versions?.node?.match(/^\d+\.\d+\.\d+/)?.[0] ?? '18'
+  return `node${version}`
```

Comment:
It looks like `esbuild` is able to accept the nightly builds version string too. I think we skip this part then